protobuf/sync/v1/sync_service.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+
+package sync.v1;
+
+option go_package = "flagd/grpcsync";
+
+/*
+   SyncFlagsRequest convey a key that identifies the flagd instance, enabling Sync provider to filter
+   specific flags for a specific flagd deployment.
+*/
+message SyncFlagsRequest {
+  string key = 1;
+}
+
+//SyncState convey the state of the payload
+enum SyncState {
+  // Value is ignored by the listening flagd
+  SYNC_STATE_UNSPECIFIED = 0;
+
+  // All the flags matching the request. This is the default response and other states can be ignored
+  // by the implementation. Flagd internally replace all existing flags for this response state.
+  SYNC_STATE_ALL = 1;
+
+  // Convey an addition of a flag. Flagd internally handles this by combining new flags with existing ones
+  SYNC_STATE_ADD = 2;
+
+  // Convey an update of a flag. Flagd internally try to update if the updated flag already exist OR if it is not,
+  // then it will get added
+  SYNC_STATE_UPDATE = 3;
+
+  // Convey a deletion of a flag. Flagd internally removes the flag
+  SYNC_STATE_DELETE = 4;
+
+  // Optional server ping to check client connectivity. Handling is ignored by flagd and is to merely support live check
+  SYNC_STATE_PING = 5;
+}
+
+// SyncFlagsResponse is the response contains flags and state
+message SyncFlagsResponse {
+  // flagd feature flag configuration
+  string flags = 1;
+  // State conveying the operation to be performed by flagd. See descriptions of state for an explanation of values
+  SyncState state = 2;
+}
+
+// FlagService implements a server streaming to provide realtime flag configurations
+service FlagService {
```

Comment:
It may be worthwhile updating this name before we implement these changes, something along the lines of `FlagSyncService` may be more appropriate, it will appear in both client and server code and may be slightly confusing if we expand this definition down the line

Reply:
/agree

Reply:
Yup agree, FlagSyncService sounds good